torchdynamo/variables/lists.py

```diff
@@ -305,6 +306,17 @@ def __init__(self, items, **kwargs):
             start, stop, step = items
         else:
             assert False
+
+        # Avoids a .item() call in the tensor slice that would attempt to get a
+        # value out fake tensors, and which would determine the output shape of
+        # the slice.  It is a workaround until
+        # https://github.com/pytorch/pytorch/pull/83567 is landed and there is
+        # more complete support for breaking on data dependent operators.
+        if not config.capture_scalar_outputs:
```

Comment:
This is correct. @voznesenskym and I briefly chatted. Basically, that flag was introduced for the Edge export usecase where they had a strict requirement of no graph break. This ensures that graph export works. With that in mind, this change is good.

Your recent issues suggest that we have started rethinking about the `capture_scalar_outputs`, so it might need some work after we get more clarity on this.

Reply:
yup, sounds good. i have other changes coming through soon that should solve this more thoroughly i think